Partial revert of 9076a631fe3 to work around a compilation error on
Guix.  The new check fails to detect Xrender even though it's present
in the compilation environment.

--- a/configure.ac
+++ b/configure.ac
@@ -3294,12 +3294,7 @@ AC_DEFUN
 # Check for XRender
 HAVE_XRENDER=no
 if test "${HAVE_X11}" = "yes"; then
-  AC_COMPILE_IFELSE(
-    [AC_LANG_PROGRAM(
-       [[#include <X11/Intrinsic.h>
-	 #include <X11/extensions/Xrender.h>
-       ]],
-       [[return !XRenderQueryExtension;]])],
+  AC_CHECK_HEADER([X11/extensions/Xrender.h],
     [AC_CHECK_LIB([Xrender], [XRenderQueryExtension], [HAVE_XRENDER=yes])])
   if test $HAVE_XRENDER = yes; then
     XRENDER_LIBS="-lXrender"
